### PR TITLE
Closes #62 advs template add interpolation functions

### LINKED ISSUE #62: ADVS template: add interpolation functions
In our ADVS template we should update the reference files with the interpolated rows so the CDC charts (>=2 yrs old) would be by day then, just like the WHO charts. The rationale for this is we already currently massage the reference files in our template, e.g. converting the CDC months into days to be more consistent with WHO charts. So this way our package still contains the raw files from CDC, but we show the users complete transparency of how we update these to make them into the metadata we pass into our function. Then if a user ever wants to simplify and remove the interpolation then they could just delete the block of code that we’d add to the template to do this. 

The interpolation will require 2 new R functions to be added to our package (one for BMI incl P95 & Sigma, and one for the other parameters without this). These 2 functions can then be called in the template Metadata section.

David has provided example code here but we should try to re-write this using tidyverse packages (preferably keeping our package dependencies lean as possible) and not data.table, and we should match the admiraldev conventions.

### PR DESCRIPTION
**Pull request by Fanny-Gautier**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: inst/WORDLIST [hunk 1]
```diff
@@ -24,3 +24,4 @@ onwards
 pharmaverse
 prm
 th
+metadataset
```
Comment (rossfarrugia) [inline inst/WORDLIST:27]: please move up to line 23 keep this file alphabetical for ease of maintenance

File: tests/testthat/test-derive_interp_records.R [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# derive_interp_records ----
```
Comment (rossfarrugia) [inline tests/testthat/test-derive_interp_records.R:1]: please add some tests for actual interpolated values to be sure the function works as expected, as currently these tests are only error checking. see https://github.com/pharmaverse/admiralpeds/pull/54/files for example

File: inst/templates/ad_advs.R [hunk 3]
```diff
@@ -44,8 +44,16 @@ bmi_for_age <- who_bmi_for_age_boys %>%
         SEX == 2 ~ "F",
         TRUE ~ NA_character_
       ),
-      AGE = AGE * 30.4375
-    )) %>%
+      AGE = round(AGE * 30.4375)
+    ) %>%
+    # Interpolate the AGE by SEX
+    group_by(SEX) %>%
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:50]: wouldn't it be cleaner to have this as `by_vars` argument within the `derive_interp_records()` function and then you could do all the group and ungroup in the function and for the template we just call something like `derive_interp_records(., parameter = "BMI", by_vars = exprs(SEX))`. at the moment this step is doing a lot and its not simple for users to follow.

File: R/derive_interp_records.R [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+#'   group_by(SEX) %>%
+#'   # Ensure first that Age unit is "DAYS"
+#'   do(derive_interp_records(., parameter = "HEIGHT")) %>%
+#'   ungroup() %>%
+#'   # Keep patients >= 2 yrs only - Remove duplicates for 730 Days old which
+#'   # must come from WHO metadata only
+#'   filter(AGE >= 730.5)
+#'
+#' print(cdc_htage)
+derive_interp_records <- function(dataset, parameter = "WEIGHT") {
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:52]:
```suggestion
derive_interp_records <- function(dataset,
                                  parameter) {
```
parameter shouldn't have a default value and then more consistent with the other functions. also if you add `by_vars` as per other comment better to show these arguments each on new line as looks better in the reference page

File: R/derive_interp_records.R [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+#'   mutate(
+#'     SEX = case_when(
+#'       SEX == 1 ~ "M",
+#'       SEX == 2 ~ "F",
+#'       TRUE ~ NA_character_
+#'     ),
+#'     AGE = round(AGE * 30.4375)
+#'   ) %>%
+#'   # Interpolate the AGE by SEX
+#'   group_by(SEX) %>%
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:43]: see my comment above - reminder to update example here also if you agree with the suggested update

File: R/derive_interp_records.R [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+#'
+#' @param parameter CDC/WHO metadata parameter
+#'
+#' *Permitted Values*: `"WEIGHT"`, `"HEIGHT"` or `"BMI"` only - Must not be `NULL`
+#'
+#'   e.g. `parameter = "WEIGHT"`,
+#'        `parameter = "HEIGHT"`,
+#'   or   `parameter = "BMI"`.
+#'
+#' @return The input dataset plus additional interpolated records
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:22]:
```suggestion
#' @return The input dataset plus additional interpolated records.
```

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@Fanny-Gautier some initial review comments for consideration. i'll be able to fully review the template again later once we have https://github.com/pharmaverse/admiralpeds/pull/60 approved and merged.

File: inst/WORDLIST (hunk 1 above)
Comment (Fanny-Gautier) [reply inst/WORDLIST]: Done

File: inst/templates/ad_advs.R (hunk 3 above)
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R]: Done

File: R/derive_interp_records.R (hunk 4 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Done

File: R/derive_interp_records.R (hunk 5 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Done

File: R/derive_interp_records.R (hunk 4 above)
Comment (rossfarrugia) [reply R/derive_interp_records.R]: @Fanny-Gautier you still left the default values:

```
derive_interp_records <- function(dataset,
                                  by_vars = exprs(SEX),
                                  parameter = "WEIGHT") {
```

it should be:

```
derive_interp_records <- function(dataset,
                                  by_vars = NULL,
                                  parameter) {
```

File: R/derive_interp_records.R [hunk 7]
```diff
@@ -40,16 +47,13 @@
 #'     AGE = round(AGE * 30.4375)
 #'   ) %>%
 #'   # Interpolate the AGE by SEX
-#'   group_by(SEX) %>%
 #'   # Ensure first that Age unit is "DAYS"
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:50]: Comment `# Ensure first that Age unit is "DAYS"` belongs above line 47

File: R/derive_interp_records.R [hunk 8]
```diff
@@ -40,16 +47,13 @@
...
-#'   do(derive_interp_records(., parameter = "HEIGHT")) %>%
-#'   ungroup() %>%
-#'   # Keep patients >= 2 yrs only - Remove duplicates for 730 Days old which
-#'   # must come from WHO metadata only
-#'   filter(AGE >= 730.5)
+#'   derive_interp_records(., parameter = "HEIGHT")
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:51]:
```suggestion
#'   derive_interp_records(., by_vars = exprs(SEX), parameter = "HEIGHT")
```
as `by_vars` should be NULL as default

File: R/derive_interp_records.R [hunk 9]
```diff
@@ -59,16 +63,33 @@ derive_interp_records <- function(dataset, parameter = "WEIGHT") {
...
+    group_by(!!!by_vars) %>%
+    do({
+      age <- .$AGE
+      x <- lapply(.[, metadata_vars], fapp, age = age)
+      as.data.frame(do.call(bind_cols, x))
+    }) %>%
+    ungroup() %>%
+    # Keep patients >= 2 yrs till 20 yrs - Remove duplicates for 730 Days old which
+    # must come from WHO metadata only
+    filter(!is.na(AGE) & AGE >= 730.5 & AGE <= 7305)
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:92]: The step for ` # Keep patients >= 2 yrs till 20 yrs - Remove duplicates for 730 Days old...` should not be in this function. Better to just leave as `filter(!is.na(AGE))`, because what about the users who don't want to follow our rules here around only using CDC growth charts for >=2 years. Our functions need to be flexible to be applied to whatever choice of metadata the user likes, e.g. maybe they only want to use interpolated CDC for all age ranges (and not WHO at all). Then in our template only we can make such recommended choices.

File: R/derive_interp_records.R [hunk 10]
```diff
@@ -59,16 +63,33 @@ derive_interp_records <- function(dataset, parameter = "WEIGHT") {
...
+
+  return(result)
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:94]: is this needed? couldn't we just use `dataset <- dataset %>%` in line 82 instead?

File: inst/templates/ad_advs.R [hunk 11]
```diff
@@ -47,13 +47,8 @@ bmi_for_age <- who_bmi_for_age_boys %>%
...
     ) %>%
     # Interpolate the AGE by SEX
-    group_by(SEX) %>%
     # Ensure first that Age unit is "DAYS"
-    do(derive_interp_records(., parameter = "BMI")) %>%
-    ungroup() %>%
-    # Keep patients >= 2 yrs only - Remove duplicates for 730 Days old which
-    # must come from WHO metadata only
-    filter(AGE >= 730.5)) %>%
+    derive_interp_records(., parameter = "BMI")) %>%
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:51]: See comments above about `by_vars` and update template in similar way. also i think styler will probably want us to write it more like: 

```
derive_interp_records(by_vars = exprs(SEX),
                      parameter = "HEIGHT")
```

I think through using `%>%` the `.,` can be avoided but tell me if i'm wrong.

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@Fanny-Gautier some further review comments from your updates

File: R/derive_interp_records.R (hunk 4 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Implemented, I misunderstood the first comment.

File: R/derive_interp_records.R (hunk 7 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R:50]: Comment moved up

File: R/derive_interp_records.R (hunk 8 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R:51]: Implemented

File: R/derive_interp_records.R (hunk 9 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R:92]: Moved back to the ADVS template

File: R/derive_interp_records.R (hunk 10 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R:94]: implemented

File: inst/templates/ad_advs.R (hunk 11 above)
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R:51]: implemented

**Comment by Fanny-Gautier** [comment]
I am currently unable to run `pkgdown::build_site()` with the default value `NULL` defined in `derive_interp_records()`
image

File: tests/testthat/test-derive_interp_records.R (hunk 2 above)
Comment (Fanny-Gautier) [reply tests/testthat/test-derive_interp_records.R:1]: it's still ongoing. I'll try to complete it by tomorrow EOB.

File: R/derive_interp_records.R (hunk 10 above)
Comment (rossfarrugia) [reply R/derive_interp_records.R]: do we need the `return(dataset)` line still now that you made this update? i thought we might be able to cut down a line of code with this update

File: R/derive_interp_records.R [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+#'   derive_interp_records(by_vars = exprs(SEX), parameter = "HEIGHT")
+#'
+#' print(cdc_htage)
+derive_interp_records <- function(dataset,
+                                  by_vars = NULL,
+                                  parameter) {
+  # Apply assertion to by_vars argument to ensure it is appropriate class
+  assert_vars(by_vars, optional = TRUE)
+
+  stopifnot(!is.null(parameter))
```
Comment (zdz2101) [inline R/derive_interp_records.R:58]:
```suggestion
  assert_character_scalar(parameter, values = c("HEIGHT", "WEIGHT", "BMI"))
```

File: R/derive_interp_records.R [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+  stopifnot(!is.na(parameter))
+  stopifnot(parameter %in% c("HEIGHT", "WEIGHT", "BMI"))
+
+  metadata_vars <- c("AGE", "L", "M", "S")
+  if (parameter == "BMI") {
+    # metadata_vars <- c("AGE", "L", "M", "S", "P95", "Sigma")
```
Comment (zdz2101) [inline R/derive_interp_records.R:64]: remove unnecessary code

File: R/derive_interp_records.R [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+    metadata_vars <- append(metadata_vars, c("P95", "Sigma"))
+  }
+
+  stopifnot("AGE" %in% colnames(dataset))
```
Comment (zdz2101) [inline R/derive_interp_records.R:68]:
```suggestion
  assert_data_frame(dataset, required_vars = exprs(!!by_vars, AGE, AGEU, L, M, S))
  if (parameter == "BMI") {
    assert_data_frame(dataset, required_vars = exprs(!!by_vars, AGE, AGEU, L, M, S, P95, Sigma))
  }
```
lines 62 - 80 can be handled with something along the lines of this

**Review by zdz2101 — COMMENTED** [review]
most of the `stopifnot()` calls can be swapped with the appropriate `admiraldev` assertions

File: inst/templates/ad_advs.R [hunk 15]
```diff
@@ -44,8 +44,17 @@ bmi_for_age <- who_bmi_for_age_boys %>%
...
+      AGE = round(AGE * 30.4375)
+    ) %>%
+    # Interpolate the AGE by SEX
+    derive_interp_records(
+      by_vars = exprs(SEX),
+      parameter = "BMI"
+    ) %>%
+    # Keep patients >= 2 yrs till 20 yrs - Remove duplicates for 730 Days old which
+    # must come from WHO metadata only
+    filter(AGE >= 730.5 & AGE <= 7305)) %>%
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:57]: do we need the `& AGE <= 7305` here? seems redundant to me, and same for other lines

Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R:57]: This is because `cdc_bmiage` metadata includes data till 240.5 months ( = 7320.22 days which is 20.04 yrs old). Should it be considered ? When David interpolated the data, he fixed the upper limit to 7305 days.

**Comment by Fanny-Gautier** [comment]
> I am currently unable to run `pkgdown::build_site()` with the default value `NULL` defined in `derive_interp_records()` image

I finally could run `pkgdown::build_site()`but do I need to call `library(admiral)` in addition to `dplyr` for the below error?
image

File: R/derive_interp_records.R (hunk 12 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Implemented

File: R/derive_interp_records.R (hunk 13 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Implemented

File: R/derive_interp_records.R (hunk 14 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Implemented

File: inst/templates/ad_advs.R (hunk 15 above)
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:57]: ok np, fine by me then and you do explain in the comment that its only up to 20 years

**Comment by rossfarrugia** [comment]
> I finally could run `pkgdown::build_site()`but do I need to call `library(admiral)` in addition to `dplyr` for the below error?

sure, add `library(admiral, warn.conflicts = FALSE)` and see if that resolves

File: tests/testthat/test-derive_interp_records.R (hunk 2 above)
Comment (Fanny-Gautier) [reply tests/testthat/test-derive_interp_records.R:1]: I have implemented them, could you please review them ? I had to add `library(admiraldev)` and `library(dplyr)` as it was failing otherwise. Could you confirm this is necessary ?
image

File: R/derive_interp_records.R (hunk 10 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: return() has now been deleted

File: tests/testthat/test-derive_interp_records.R (hunk 2 above)
Comment (rossfarrugia) [reply tests/testthat/test-derive_interp_records.R:1]: @zdz2101 left the following comment on Laura's PR if it helps:

Per [textbook](https://r-pkgs.org/testing-basics.html#test-organisation) you'll want to avoid calling libraries in unit-test suite, if you need to use {pkgname} functions just use pkgname::xxx() format, demonstrated below with the tibble::tribble() call. Although it's ideal to this with all outside-functions, I think dplyr ones are so commonly used we can use the importFrom in the admiralpeds-package.R to avoid excessive use

Also can you merge latest main and resolve the conflict with `R/admiralpeds-package.R`

**Comment by Fanny-Gautier** [comment]
@zdz2101 , @rossfarrugia : I am still facing the error with `by_vars` since using `assert_vars()` in the example.
Could you please suggest any solution ? I have unsuccessfully worked around the whole afternoon.
image
image
Thanks.

All other comments are implemented. Please let me know if there are further comments.

File: tests/testthat/test-derive_interp_records.R (hunk 2 above)
Comment (Fanny-Gautier) [reply tests/testthat/test-derive_interp_records.R:1]: Implemented. Thanks for the guidance.

File: R/derive_interp_records.R [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+#'     by_vars = exprs(SEX),
+#'     parameter = "HEIGHT"
+#'   )
+#'
+#' print(cdc_htage)
+derive_interp_records <- function(dataset,
+                                  by_vars = NULL,
+                                  parameter) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  by_vars <- assert_vars(by_vars, optional = TRUE)
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:59]: why is the `by_vars <- ` needed here?

File: R/derive_interp_records.R [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+  assert_character_scalar(parameter, values = c("HEIGHT", "WEIGHT", "BMI"))
+  assert_data_frame(dataset, required_vars = exprs(!!!by_vars, AGE, L, M, S))
+  if (parameter == "BMI") {
+    assert_data_frame(dataset, required_vars = exprs(!!!by_vars, AGE, L, M, S, P95, Sigma))
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:63]: in the above 2 `assert_data_frame()` calls we have `!!!by_vars` as required, but then `by_vars=NULL` is allowed in this function

File: R/derive_interp_records.R [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  arrange(dataset, !!!by_vars, AGE)
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:66]: what about when `by_vars=NULL`? shouldn't this sort depend on whether `by_vars` is not NULL?

File: R/derive_interp_records.R [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  # Linear interpolation
+  fapp <- function(v, age) {
+    approx(age, v, xout = seq(min(age), max(age)))$y
+  }
+
+  # Apply the function within each group and combine the results
+  dataset <- dataset %>%
+    group_by(!!!by_vars) %>%
```
Comment (rossfarrugia) [inline R/derive_interp_records.R:97]: what about when `by_vars=NULL`? shouldn't this step depend on whether `by_vars` is not NULL?

**Comment by rossfarrugia** [comment]
> @zdz2101 , @rossfarrugia : I am still facing the error with `by_vars` since using `assert_vars()` in the example. Could you please suggest any solution ? I have unsuccessfully worked around the whole afternoon. 

@bundfussr can you take a look at this PR please and the error when we run `pkgdown::build_site()`. It's something to do with the assertions as `by_vars` is allowed to be NULL. Appreciate you sanity checking as you'd probably find the issue a million times quicker than me.

File: R/derive_interp_records.R (hunk 17 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R:63]: it works fine while `by_vars = NULL`: Tested in the ADVS template

File: R/derive_interp_records.R (hunk 18 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Loop imlemented

File: R/derive_interp_records.R (hunk 19 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Loop implemented

File: R/derive_interp_records.R (hunk 16 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Dropped

**Comment by bundfussr** [comment]
> > @zdz2101 , @rossfarrugia : I am still facing the error with `by_vars` since using `assert_vars()` in the example. Could you please suggest any solution ? I have unsuccessfully worked around the whole afternoon.
> 
> @bundfussr can you take a look at this PR please and the error when we run `pkgdown::build_site()`. It's something to do with the assertions as `by_vars` is allowed to be NULL. Appreciate you sanity checking as you'd probably find the issue a million times quicker than me.

Adding `library(rlang)` at the beginning should solve the error when we run `pkgdown::build_site()`.

**Comment by bundfussr** [comment]
By the way, superseded dplyr functions like `group_by_at()` and `do()` are used in the code. I wonder if this should be avoided.

File: R/derive_interp_records.R (hunk 17 above)
Comment (bundfussr) [reply R/derive_interp_records.R:64]: This shouldn't matter. If `by_vars = NULL`, `exprs(!!!by_vars, AGE, L, M, S)` is the same as `exprs(AGE, L, M, S))`. I.e., `NULL` arguments are ignored.

File: R/derive_interp_records.R (hunk 18 above)
Comment (bundfussr) [reply R/derive_interp_records.R]: Same as above. `NULL` arguments are ignored.

**Comment by bundfussr** [comment]
And to be in line with the admiral messaging cli functions should be used for messages, e.g., `cli_abort()` instead of `stop()`.

File: R/derive_interp_records.R (hunk 19 above)
Comment (bundfussr) [reply R/derive_interp_records.R]: I think even this should work for `by_vars = NULL`.

File: R/derive_interp_records.R (hunk 18 above)
Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: Same here, it was working fine with `by_vars = NULL`. I have implemented a loop to make it clear even if it was working fine without the loop

File: R/derive_interp_records.R (hunk 17 above)
Comment (rossfarrugia) [reply R/derive_interp_records.R:64]: ah ok sorry, my lack of understanding! sorry @Fanny-Gautier - i just wanted to check that these still worked for `by_vars=NULL`, that was my main comment. please do put these back to your simpler way of having this.

File: R/derive_interp_records.R (hunk 19 above)
Comment (bundfussr) [reply R/derive_interp_records.R]: Maybe a unit test for `by_vars = NULL` should be added. The coverage is quite low (57.89%) at the moment.

Comment (rossfarrugia) [reply R/derive_interp_records.R]: R is so much cleverer than the old programming language i worked in 😄

**Comment by rossfarrugia** [comment]
@Fanny-Gautier appreciate if you can look into Stefan's 2 comments above and undo my bad suggestions above - sorry 😞 and then hopefully we can get this one merged!

File: R/derive_interp_records.R (hunk 19 above)
Comment (rossfarrugia) [reply R/derive_interp_records.R]: agree on the extra unit test too!

Comment (Fanny-Gautier) [reply R/derive_interp_records.R]: I'll implement my checks in the unit tests - thanks for the suggestion

**Comment by Fanny-Gautier** [comment]
Could you please let me know how to handle the below "error" ?
When running styler, it set back my lines to 6 spaces, while Lintr requires 21 spaces. So when it fits to Lintr this makes a styler error, and when it fits to styler, it makes a Lintr error.
image

**Comment by bundfussr** [comment]
> Could you please let me know how to handle the below "error" ? When running styler, it set back my lines to 6 spaces, while Linter requires 21 spaces. So when it fits to Linter this makes a styler error, and when it fits to styler, it makes a Linter error. image

Yes, this is a known issue. Please add `indentation_linter=NULL,` to the `.lintr` file. This should solve the issue.

**Review by zdz2101 — APPROVED** [review]
LGTM!

**Comment by rossfarrugia** [comment]
@Fanny-Gautier i ran through the ADVS template and the results look good. i also checked test coverage using covr and your new function is 100% covered. so i'll approve/merge this now so that i can merge into my open PR.

@bundfussr if you spot any further code improvements for this one please can you make an issue for us and we can tackle separately to this PR.

**Review by rossfarrugia — APPROVED** [review]
